File: FizzBuzz/FizzBuzz.swift
// run with Xcode 7.2(7C46l)

let fizzBuzz: [(num: Int, word: String)] = [(3, "Fizz"), (5, "Buzz")]

for i in 1...100 {
    let result = fizzBuzz.reduce("") {str, fb in
        str + (i % fb.num == 0 ? fb.word : "")
    }
    print(result != "" ? result : i)
}

File: FizzBuzz/FizzBuzz.rb
(1..100).each do |i|
  case
    when i % 15 == 0
      puts "FizzBuzz"
    when i % 5 == 0
      puts "Buzz"
    when i % 3 == 0
      puts "Fizz"
    else
      puts i
  end
end
